065-topic-replication-factor-change.md

```diff
@@ -0,0 +1,290 @@
+The `goals` parameter is a comma separated list of goals used to generate the automatic cluster rebalance for the replication factor change.
+This is configurable for all endpoints in `Kafka.spec.cruiseControl.config` using the `default.goals` property.
+
+The `replication_throttle` parameter is an upper bound on the bandwidth used to move replicas (bytes per second).
+This is configurable for all endpoints in `Kafka.spec.cruiseControl.config` using the `default.replication.throttle` property.
+
+### User tasks
+
+The [`user_tasks`](https://github.com/linkedin/cruise-control/wiki/REST-APIs#query-the-user-request-result) endpoint is used to periodically check the replicas change tasks state.
+The only required parameter is the `User-Task-ID`, which is returned by the `topic_configuration` response.
+A completed task is retained for `completed.user.task.retention.time.ms`, which defaults to 24 hours.
+
+```sh
+$ curl -s "localhost:9090/kafkacruisecontrol/user_tasks?user_task_ids=5344ca89-351f-4565-8d0f-9aade00e053d&json=true" | jq
+{
+  "userTasks": [
+    {
+      "Status": "Completed",
+      "ClientIdentity": "127.0.0.1",
+      "RequestURL": "POST /kafkacruisecontrol/topic_configuration?topic=my-topic&replication_factor=2&skip_rack_awareness_check=true&dryrun=false&json=true",
+      "UserTaskId": "5344ca89-351f-4565-8d0f-9aade00e053d",
+      "StartMs": "1701961135978"
+    }
+  ],
+  "version": 1
+}
+```
+
+### Configuration parameters
+
+The Topic Operator needs to authenticate and send HTTP requests to the Cruise Control's REST API.
+The following environment variables are set by the Cluster Operator for consumption by the Topic Operator when Cruise Control is configured in Kafka resource, or manually by the user in case of standalone deployment:
+
+| Name                                  | Type    | Default | Description                                                   |
+|:--------------------------------------|---------|---------|---------------------------------------------------------------|
+| STRIMZI_CRUISE_CONTROL_ENABLED        | Boolean | false   | Whether Cruise Control integration is enabled                 |
+| STRIMZI_CRUISE_CONTROL_RACK_ENABLED   | Boolean | false   | Whether rack awareness is enabled in the target Kafka cluster |
+| STRIMZI_CRUISE_CONTROL_HOSTNAME       | String  | ""      | Cruise Control hostname                                       |
+| STRIMZI_CRUISE_CONTROL_PORT           | Integer | 9090    | Cruise Control port                                           |
+| STRIMZI_CRUISE_CONTROL_SSL_ENABLED    | Boolean | false   | Whether SSL encryption is enabled                             |
+| STRIMZI_CRUISE_CONTROL_AUTH_ENABLED   | Boolean | false   | Whether Basic authentication is enabled                       |
+
+With the Cluster Operator deployment, the certificate chain in PEM format and the admin user credentials for the Cruise Control REST API are automatically mounted in the Topic Operator container.
```

Comment:
This is separate from the credentials for the `KafkaRebalanceOperator` right? So we would have two different admin users:
(1) KafkaRebalanceOperator
(2) UTO
It would probably be easier to implement this way and we would also know which component made which CC API request soley by the user name when looking at the CC UserTask list

Reply:
Yes, it's a dedicated user. Let me clarify in the proposal.